src/main/java/com/example/ludogorieSoft/village/services/VillageService.java

```diff
@@ -22,21 +32,44 @@ public class VillageService {
     private final VillageRepository villageRepository;
     private final ModelMapper modelMapper;
     private final RegionService regionService;
+    private final AuthService authService;
     private static final String ERROR_MESSAGE1 = "Village with id ";
     private static final String ERROR_MESSAGE2 = " not found  ";
 
 
     public VillageDTO villageToVillageDTO(Village village) {
         return modelMapper.map(village, VillageDTO.class);
     }
+    public VillageResponse villageToVillageResponse(Village village){
+        return modelMapper.map(village, VillageResponse.class);
+    }
 
     public List<VillageDTO> getAllVillages() {
         List<Village> villages = villageRepository.findAll();
         return villages.stream()
                 .map(this::villageToVillageDTO)
                 .toList();
     }
+    public List<VillageResponse> getAllVillagesWithAdmin() {
```

Comment:
What is the idea of this method? Seems that instead of iterating over the villages you can make a query that returns the desired result. With query it would be faster for the server.

Also the name of the method is not correct. It returns collection of VillageResponses and according to name it returns Village.

Reply:
this method is old before changes, it is not using now, I will delete it